test/nuts/local/partialBundleDelete.nut.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as path from 'path';
+import * as fs from 'fs';
+import { TestSession } from '@salesforce/cli-plugins-testkit';
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import { getComponentSets } from '../../../src/shared/localComponentSetArray';
+
+describe('Bundle-like types delete', () => {
+  let session: TestSession;
+
+  before(async () => {
+    session = await TestSession.create({
+      project: {
+        sourceDir: path.join('test', 'nuts', 'repros', 'partialBundleDelete'),
+      },
+      authStrategy: 'NONE',
+    });
+  });
+
+  // We need a sinon sandbox to stub the file system to make it look like we
+  // deleted some files.
+  const sandbox = sinon.createSandbox();
+
+  after(async () => {
+    await session?.clean();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('returns components for deploy with partial LWC delete', () => {
+    const lwcTestCompDir = path.join(session.project.dir, 'force-app', 'lwc', 'testComp');
+    const lwcHtmlFile = path.join(lwcTestCompDir, 'myComp.html');
+    const lwcJsFile = path.join(lwcTestCompDir, 'myComp.js');
+    const lwcMetaFile = path.join(lwcTestCompDir, 'myComp.js-meta.xml');
+
+    const compSets = getComponentSets([
+      {
+        path: path.join(session.project.dir, 'force-app', 'lwc'),
+        nonDeletes: [lwcJsFile, lwcMetaFile],
+        deletes: [lwcHtmlFile],
+      },
+    ]);
+
+    expect(compSets.length).to.equal(1);
+    compSets.forEach((cs) => {
+      expect(cs.getTypesOfDestructiveChanges()).to.deep.equal([]);
+      const comps = cs.getSourceComponents().toArray();
+      expect(comps[0].isMarkedForDelete()).to.equal(false);
+    });
+  });
+
+  it('returns components for delete for full LWC delete', () => {
+    // We stub this so it appears that we deleted all the LWC files
```

Comment:
alternative to sinon:  delete stuff.

And then if you *do* need those files back , you could always copy them back from the repros folder